### ISSUE TITLE
How to train or test on the Kitti dataset?

### ISSUE THREAD

**Issue by WERush** [tinghuiz/appearance-flow#5]
Hi, Can you give me some instructions on how to train or test on the Kitti dataset?

Has the 'ObjRenderer' been used when training on the Kitti dataset? I saw it was designed for shapenet dataset.

Thanks.

**Comment by gunshi** [comment]
+1

**Comment by tinghuiz** [comment]
The ObjRender was not used for training KITTI. Please refer to Sec. 4.2 of the paper on how the training/testing data were generated.

**Comment by gunshi** [comment]
Hey, in my case i meant to ask if you would be releasing Kitti training scripts.
(I know ObjRender is used only for shapenet, sorry if it sounded like i was asking the same question)
Thanks!

**Comment by tinghuiz** [comment]
I might release the training scripts in the future, but definitely not any time soon.

**Comment by gunshi** [comment]
Okay.
I was interested in training and seeing the results, so I made my implementation at [appearance-flow-tensorflow](https://github.com/gunshi/appearance-flow-tensorflow) (If anyone is interested)
I've only trained for a few epochs but my predicted images are not very close to target images, though that may be due to some hyperparameter differences though i've tried to keep everything the same.


I had one question, for the multi view case for KITTI, in the demo on the github page there are two frames (red and green for the two input views), and the intermediate frames are interpolated. In that scenario have you taken frames '1' and say, 'n' as input, and to interpolate frame say, '3', given 'tforms' 1->3 and n->3 as input transforms  ?

**Comment by tinghuiz** [comment]
This is great! I have been wanting to do a tensorflow implementation myself. Now I can just help you with it :) 
Can you send some examples of your predicted images? Btw, tensorflow has a native op for bilinear sampling. You might want to check out [this](https://www.tensorflow.org/api_docs/python/tf/contrib/resampler/resampler).

I think you are right. The input transforms of the intermediate frames are interpolated. But I'm not completely sure as it was done a long time ago...

**Comment by gunshi** [comment]
Oh that's great.
I hadn't seen the resampler, probably because its quite recent. Will try to integrate that instead of my own implementation, and also the ShapeNet code, so that this works for both KITTI and shapenet.

Images i'll send soon, as soon as i'm back from traveling.

**Comment by gunshi** [comment]
Hey @tinghuiz ! I'm sorry I got busy with something else, and wanted to first look into all the code again to see where i might be going wrong.
Yeah so I ran it again with batch size 30, saving it after every 2000 iterations, and so far till the 4000th(1,20,000 images later) iteration it still doesn't give meaningful results. At least visually inspecting the images makes it look like a common leftward horizontal shift is being applied to all images.
dropbox link to results: https://www.dropbox.com/sh/eq2wq4qxlcc45fv/AAAiwXD7ARccCsQUdckljuOna?dl=0
Is it that I'm training it for long enough?
Thanks,
Gunshi
